src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -377,11 +378,11 @@ private IEnumerable<ServerPackage> CachedPackages
         {
             get
             {
-                if (!_serverPackageStore.HasPackages())
+                if (_needsRebuild || !_serverPackageStore.HasPackages())
```

Comment:
nit: Add a comment to explain this, it makes perfect sense after reading your PR description. But initially I thought the purpose here was to make it always run the first time.